### PR TITLE
i#4318 xarch memtrace: Add README for testdata

### PR DESCRIPTION
**Pull request by derekbruening**
Adds a README about how to re-generate the testdata added in PR #4330
commit 58a911a1.

Issue: #4318

### CODE REVIEW

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README [hunk 1]
```diff
@@ -0,0 +1,50 @@
+How to re-generate the drmemtrace.threadsig.aarch64.raw files
+
+The application run is "threadsig" from https://github.com/DynamoRIO/drmemtrace_samples/blob/master/threadsig.aarch64/threadsig.cp
```
Comment (abhinav92003) [inline clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:3]: threadsig.cpp (missed one letter at the end)

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+To save repository space, build it with -Os and -static on an AArch64
```
Comment (abhinav92003) [inline clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:5]: There is a binary present in the repo: https://github.com/DynamoRIO/drmemtrace_samples/blob/master/threadsig.aarch64/binaries/threadsig, which is same as the one checked in at dynamorio/clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/threadsig. Is it ok to just use that one? Or should I recompile and regenerate the trace?

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  $ gzip `ls -1td drmemtrace.threadsig.*.dir|head -1`/raw/*.raw
+
+Now copy the new data files and the threadsig and libdynamorio.so.
+
+Strip libdynamorio.so, though this may not decrease it much:
+
+  $ strip lib64/release/libdynamorio.so
+
+There is no need to change the checked-in libdrmemtrace.so.  It is not
```
Comment (abhinav92003) [inline clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:26]: Curious why dynamorio includes it in the list of modules if it is not executed?

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+executed.  I created it by just copying a pure-asm tiny ELF binary:
+e.g., suite/tests/bin/common.allasm_aarch64_isa.
+
+A smaller libdynamorio.so I created by compiling this file
```
Comment (abhinav92003) [inline clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:30]: The part below seems optional. Perhaps, mark it as such?

**Review by abhinav92003 — APPROVED** [review]

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README (hunk 3 above)
Comment (derekbruening) [reply clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:26]: You mean, why is a client library in the list of app libraries?  And why just the top-level client and none of the extension libraries it uses?  Some client libraries do have the app execute some of their code (such as when using drwrap_replace).  The client lib is mapped early, and its deps loaded later, and the memory walk finds it, and we deemed it better to include it than omit it for the code reason.  There is an issue on it IIRC.

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README (hunk 2 above)
Comment (derekbruening) [reply clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:5]: Hmm, yeah, just running that one seems fine.  I will update the README to give that option.

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README (hunk 4 above)
Comment (derekbruening) [reply clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:30]: It does save 300K so I was waffling whether to push it.  WDYT?

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README (hunk 2 above)
Comment (abhinav92003) [reply clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:5]: The output binary after following these steps is exactly the same.

File: clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README (hunk 4 above)
Comment (derekbruening) [reply clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/README:30]: Probably it's too fiddly and complex to synthesize (have to go find dynamorio_sigreturn address, etc.) or make a tool; will mark optional and we will live with the 300K.

**Pull request merged**